deposit-app/src/main/java/edu/unc/lib/boxc/deposit/validate/ValidateContentModelJob.java

```diff
@@ -207,10 +210,27 @@ private void validateFileObject(Resource resc, Resource parentResc) {
         }
 
         Resource origResc = DepositModelHelpers.getDatastream(resc);
-        if (origResc == null) {
-            failJob(null, "No original datastream for file object {0}", resc.getURI());
-        } else if (!origResc.hasProperty(CdrDeposit.stagingLocation)) {
-            failJob(null, "No staging location provided for file ({0})", resc.getURI());
+        var hasOrigResc = origResc != null;
+
+        if (hasAtLeastOneStreamingProperty(resc)) {
+            if (!hasAllStreamingProperties(resc)) {
+                // if one streaming property is present, they must all be present
+                failJob(null, "All streaming properties are required for file ({0})", resc.getURI());
+            }
+        } else if (hasOrigResc) {
```

Comment:
At the moment if there were streaming properties and an original resc, it wouldn't check that there was a staging location. I'd suggest changing this from an `else if` to a series of `if` blocks, and adding a boolean variable like `var hasStreamingOrOriginal = false` and then setting it to true in both the hasOrigResc and hasStreaming blocks, and then using that variable to decide if we want to do the "no streaming or original" error.